sdk/src/store.rs

```diff
@@ -1386,26 +1388,76 @@ impl Store {
 
         Ok(hashes)
     }
+
+    // move or copy data from source to dest
+    #[cfg(feature = "file_io")]
+    fn move_or_copy(source: &Path, dest: &Path) -> Result<()> {
+        // copy temp file to asset
+        std::fs::rename(source, dest)
+            // if rename fails, try to copy in case we are on different volumes or output does not exist
+            .or_else(|_| std::fs::copy(source, dest).and(Ok(())))
+            .map_err(Error::IoError)
+    }
+
+    // copy output and possibly the external manifest to final destination
+    #[cfg(feature = "file_io")]
+    fn copy_c2pa_to_output(source: &Path, dest: &Path, remote_type: RemoteManifest) -> Result<()> {
+        match remote_type {
+            crate::claim::RemoteManifest::NoRemote => Store::move_or_copy(source, dest)?,
+            crate::claim::RemoteManifest::SideCar | crate::claim::RemoteManifest::Remote(_) => {
+                // make correct path names
+                let source_asset = source;
+                let source_cai = source_asset.with_extension(MANIFEST_STORE_EXT);
+                let dest_cai = dest.with_extension(MANIFEST_STORE_EXT);
+
+                Store::move_or_copy(&source_cai, &dest_cai)?; // copy manifest
+                Store::move_or_copy(source_asset, dest)?; // copy asset
```

Comment:
potential case where first copy works and second fails, would leave only half of the data moved, should we delete it in that case? Don't need to do anything now, just something to think about.

Reply:
👍  That is always an issue with move APIs.  We don't have transactions.